### **Chapter4 반복문 - 코딩 연습**

### Q4-1. 코딩 연습 : 5의 배수가 아닌 수의  합계 구하기

문제 : 다음은 키보드로 정부 범위의 시작 수와 끝 수를 입력받아 그 범위에 있는 수 중에서 5의 배수가 아닌 수의 합계를 구하는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>시작 수를 입력하세요 : 100</p>
<p>끝 수를 입력하세요 : 200</p>
<p>---------------------------------------------------------</p>
<p>100에서 200까지의 5의 배수가 아닌 수의 합계 : 12000</p>

In [1]:
n1 = int(input('시작 수를 입력하세요 : '))
n2 = int(input('끝 수를 입력하세요 : '))

sum = 0

for i in range(n1, n2 + 1) :
    if i % 5 != 0 :
        sum += i

print('-' * 50)
print('%d에서 %d까지 5의 배수가 아닌 수의 합계 : %d' % (n1, n2, sum))

시작 수를 입력하세요 : 100
끝 수를 입력하세요 : 200
--------------------------------------------------
100에서 200까지 5의 배수가 아닌 수의 합계 : 12000


### Q4-2 코딩 연습 : 전화번호에서 하이픈(-) 삭제하기

문제 : 다음은 하이픈(-)이 포함된 휴대폰 번호를 입력받아 하이픈을 삭제하는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>하이픈(-)을 포함한 휴대폰 번호를 입력하세요 : 010-1234-5678</p>
<p>01012345678</p>

In [ ]:
number = input('하이픈(-)을 포함한 휴대폰 번호를 입력하세요 : ')

for x in number :                   # 문자열 하나씩 순회하면서 검사
    if x != '-' :
        print('%s' % x, end = '')

### Q4-3 코딩 연습 : 전화번호에 하이픈(-) 추가하기

문제 : 다음은 "###########" 형태의 11자리의 휴대폰 번호를 입력받아 하이픈(-)을 추가한 "###-####-####"의 형태로 전화번호를 출력하는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>하이픈(-)을 뺀 11자리의 휴대폰 번호를 입력하세요 : 01033334444</p>
<p>010-3333-4444</p>

In [2]:
phone = input('하이픈(-)을 뺀 11자리의 휴대폰 번호를 입력하세요 : ')

number = '';

for i in range(0, len(phone)) :
    if i == 2 :
        number = number + (phone[2] + '-')
    elif i == 6 :
        number = number + (phone[6] + '-')
    else :
        number = number + phone[i]

print(number)

하이픈(-)을 뺀 11자리의 휴대폰 번호를 입력하세요 : 01033334444
010-3333-4444


### Q4-4 코딩 연습 : 길이 단위 환산표 만들기

문제 : 다음은 길이 단위(센티미티, 인치, 피트, 야드)에 대한 환산표를 만드는 프로그램이다. 밑줄 친 부분을 채우시오.

<p> * 환산 공식</p>
<p>inch = cm x 0.393701</p>
<p>ft = cm x 0.032808</p>
<p>yd = cm x 0.010936</p>

<p><실행 결과></p>
<p>-------------------------------------------------------</p>
<p>센티미터(cm) 인치(inch) 피트(ft) 야드(yd)</p>
<p>-------------------------------------------------------</p>
<p>      10        3.9       0.3       0.1</p>
<p>      20        7.9       0.7       0.2</p>
<p>      ...</p>
<p>      190       74.8      6.2       2.1</p>
<p>      200       78.7      6.6       2.2</p>
<p>-------------------------------------------------------</p>



In [3]:
print('-' * 50)
print('센티미터(cm) 인치(inch)  피트(ft)  야드(yd)')
print('-' * 50)

for cm in range(10, 201, 10) :
    inch = cm * 0.393701
    ft = cm * 0.032808
    yd = cm * 0.010936
    print('%8d %10.1f %10.1f %10.1f' % (cm, inch, ft, yd))

print('-' * 50)

--------------------------------------------------
센티미터(cm) 인치(inch)  피트(ft)  야드(yd)
--------------------------------------------------
      10        3.9        0.3        0.1
      20        7.9        0.7        0.2
      30       11.8        1.0        0.3
      40       15.7        1.3        0.4
      50       19.7        1.6        0.5
      60       23.6        2.0        0.7
      70       27.6        2.3        0.8
      80       31.5        2.6        0.9
      90       35.4        3.0        1.0
     100       39.4        3.3        1.1
     110       43.3        3.6        1.2
     120       47.2        3.9        1.3
     130       51.2        4.3        1.4
     140       55.1        4.6        1.5
     150       59.1        4.9        1.6
     160       63.0        5.2        1.7
     170       66.9        5.6        1.9
     180       70.9        5.9        2.0
     190       74.8        6.2        2.1
     200       78.7        6.6        2.2
------------------------

### Q4-5 코딩 연습 : 무게 단위 환산표 만들기

문제 : 다음은 길이 단위(킬로그램(kg), 파운드(lb), 온스(oz)에 대한 환산표를 만드는 프로그램이다. 밑줄 친 부분을 채우시오.

<p>* 환산 공식</p>
<p>lb = kg x 2.204623</p>
<p>oz = kg x 35.273962</p>

<p><실행 결과></p>
<p>--------------------------------------------------</p>
<p>킬로그램(kg) 파운드(lb)  온스(oz)</p>
<p>--------------------------------------------------</p>
<p>    10        22.0        352.7</p>
<p>    15        33.1        529.1</p>
<p>    20        44.1        705.5</p>
<p>    ...</p>
<p>    95        209.4       3351.0</p>
<p>    100       220.5       3527.4</p>
<p>--------------------------------------------------</p>

In [4]:
print('-' * 40)
print('킬로그램(kg) 파운드(lb)  온스(oz)')
print('-' * 40)

for kg in range(10, 101, 5) :
    lb = kg * 2.204623
    oz = kg * 35.273962
    print('%8d  %10.1f  %10.1f' % (kg, lb, oz))

print('-' * 40)

----------------------------------------
킬로그램(kg) 파운드(lb)  온스(oz)
----------------------------------------
      10        22.0       352.7
      15        33.1       529.1
      20        44.1       705.5
      25        55.1       881.8
      30        66.1      1058.2
      35        77.2      1234.6
      40        88.2      1411.0
      45        99.2      1587.3
      50       110.2      1763.7
      55       121.3      1940.1
      60       132.3      2116.4
      65       143.3      2292.8
      70       154.3      2469.2
      75       165.3      2645.5
      80       176.4      2821.9
      85       187.4      2998.3
      90       198.4      3174.7
      95       209.4      3351.0
     100       220.5      3527.4
----------------------------------------


### Q4-6 코딩 연습 : 홀수 개수 세기

문제 : 다음은 입력받은 숫자에 포함된 홀수의 개수를 세는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>숫자를 입력하세요 : 3049894</p>
<p>입력된 숫자 중 홀수의 개수 : 3개</p>

In [5]:
number = input('숫자를 입력하세요 : ')

total = 0

for a in number :
    a = int(a)
    if a % 2 == 1 :
        total += 1

print('입력된 숫자 중 홀수의 개수 : %d개' % total)

숫자를 입력하세요 : 3049894
입력된 숫자 중 홀수의 개수 : 3개


### Q4-7 코딩 연습 : 별표(*)로 특정 형태 만들기 1

문제 : 다음은 이중 for문을 이용하여 실행 결과가 같은 형태로 출력하는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>*</p>
<p>**</p>
<p>***</p>
<p>****</p>
<p>*****</p>
<p>******</p>
<p>*******</p>
<p>********</p>
<p>*********</p>
<p>**********</p>

In [6]:
for i in range(1, 11) :
    for j in range(1, i + 1) :
        print('*', end = '')
    print()

*
**
***
****
*****
******
*******
********
*********
**********


### Q4-8 코딩 연습 : 별표(*)로 특정 형태 만들기 2

문제 : 다음은 이중 for문을 이용하여 실행 결과가 같은 형태로 출력하는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>         *</p>
<p>        **</p>
<p>       ***</p>
<p>      ****</p>
<p>     *****</p>
<p>    ******</p>
<p>   *******</p>
<p>  ********</p>
<p> *********</p>
<p>**********</p>


In [7]:
for i in range (1,11) :
    for j in range(1, 10-i+1) :
        print(' ', end ='')
    for k in range(1, i + 1) :
        print('*', end = '')
    print()

         *
        **
       ***
      ****
     *****
    ******
   *******
  ********
 *********
**********


### Q4-9 코딩 연습 : 섭씨/화씨 환산표 만들기(while문)

문제 : 다음은 while문을 이용하여 섭씨 온도를 화씨 온도로 환산하는 표를 만드는 프로그램이다. 밑줄 친 부분을 채우시오.

<p><실행 결과></p>
<p>-------------------------------</p>
<p>    섭씨   화씨</p>
<p>-------------------------------</p>
<p>    -20    -4.0</p>
<p>    -19    -2.2</p>
<p>    -18    -0.4</p>
<p>    ...</p>
<p>     39    102.2</p>
<p>     40    104.0</p>
<p>-------------------------------</p>



In [8]:
print('-' * 30)
print('    섭씨    화씨')
print('-' * 30)

c = -20
while c < 41 :
    f = c * 9.0/5.0 + 32.0
    print('%8d %8.1f' % (c, f))

    c += 1

print('-' * 30)

------------------------------
    섭씨    화씨
------------------------------
     -20     -4.0
     -19     -2.2
     -18     -0.4
     -17      1.4
     -16      3.2
     -15      5.0
     -14      6.8
     -13      8.6
     -12     10.4
     -11     12.2
     -10     14.0
      -9     15.8
      -8     17.6
      -7     19.4
      -6     21.2
      -5     23.0
      -4     24.8
      -3     26.6
      -2     28.4
      -1     30.2
       0     32.0
       1     33.8
       2     35.6
       3     37.4
       4     39.2
       5     41.0
       6     42.8
       7     44.6
       8     46.4
       9     48.2
      10     50.0
      11     51.8
      12     53.6
      13     55.4
      14     57.2
      15     59.0
      16     60.8
      17     62.6
      18     64.4
      19     66.2
      20     68.0
      21     69.8
      22     71.6
      23     73.4
      24     75.2
      25     77.0
      26     78.8
      27     80.6
      28     82.4
      29     84.2
      30     86.0
      3

### Q4-10 코딩 연습 : 3의 배수가 안닌 수 출력하기(while문)

문제 : 다음은 while문을 이용하여 200~600의 수 중 3의 배수가 아닌 수를 출력하는 프로그램이다. 밑줄 친 부분을 채우시오.

<p>* 조건 한 줄에 8개씩 출력함</p>

<p><실행 결과></p>
<p>200 202 203 205 206 208 209 211</p>
<p>212 214 215 217 218 220 221 223</p>
<p>224 226 227 229 230 232 233 235</p>
<p>...</p>
<p>584 586 587 589 590 592 593 595</p>
<p>596 598 599</p>

In [9]:
i = 200
count = 0
while i < 601 :
    if i % 3 != 0 :
        print(' %d' % i, end='')
        count += 1
        if count % 8 == 0 :
            print()
    i += 1

 200 202 203 205 206 208 209 211
 212 214 215 217 218 220 221 223
 224 226 227 229 230 232 233 235
 236 238 239 241 242 244 245 247
 248 250 251 253 254 256 257 259
 260 262 263 265 266 268 269 271
 272 274 275 277 278 280 281 283
 284 286 287 289 290 292 293 295
 296 298 299 301 302 304 305 307
 308 310 311 313 314 316 317 319
 320 322 323 325 326 328 329 331
 332 334 335 337 338 340 341 343
 344 346 347 349 350 352 353 355
 356 358 359 361 362 364 365 367
 368 370 371 373 374 376 377 379
 380 382 383 385 386 388 389 391
 392 394 395 397 398 400 401 403
 404 406 407 409 410 412 413 415
 416 418 419 421 422 424 425 427
 428 430 431 433 434 436 437 439
 440 442 443 445 446 448 449 451
 452 454 455 457 458 460 461 463
 464 466 467 469 470 472 473 475
 476 478 479 481 482 484 485 487
 488 490 491 493 494 496 497 499
 500 502 503 505 506 508 509 511
 512 514 515 517 518 520 521 523
 524 526 527 529 530 532 533 535
 536 538 539 541 542 544 545 547
 548 550 551 553 554 556 557 559
 560 562 5

In [10]:
print('hello world')

hello world
